include/yaml-cpp/node/convert.h

```diff
@@ -17,9 +17,8 @@
 #include <type_traits>
 #include <valarray>
 #include <vector>
-#include <version>
 
-#ifdef __cpp_lib_string_view
+#if __cplusplus >= 201703L
```

Comment:
```suggestion
#if ((defined(_MSVC_LANG) && _MSVC_LANG >= 201703L) || __cplusplus >= 201703L)
```

Reply:
after reading https://learn.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=msvc-170, i still don't understand why your proposal is better. what exactly problem are you trying to address?

Reply:
Agreed. @0xFireWolf I'm going to merge this PR as-is, and if you want to suggest a change, please open a new PR with an explanation.

Reply:
Sorry, I didn't see the long conversation on #1222. I finally read that, and I understand this suggestion, @0xFireWolf, and I'd accept this as a PR with a link to the explanation in #1222.